### PR TITLE
Replace gulp-open with custom open implementation

### PR DESCRIPTION
**Pull request by AJIXuMuK**
## Summary
Replace old `gulp-open` with custom `open` implenetation

## Details

The PR removes `gulp-open` from `gulp-core-build-serve` as the library is old and not maintained. 
Currently it leads to failures in `gulp serve` command when running on WSL2.
The new `open` function mimics behavior of `gulp-open`.

## How it was tested

Manual tests by copying `lib` folder to scaffolded spfx solution

### CODE REVIEW

File: core-build/gulp-core-build-serve/src/utilities/GulpOpen.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import * as colors from 'colors';
```
Comment (iclanton) [inline core-build/gulp-core-build-serve/src/utilities/GulpOpen.ts:4]: Missing copyright.

File: core-build/gulp-core-build-serve/src/utilities/GulpOpen.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+export function open(opts: { uri?: string }): NodeJS.WritableStream {
+  /* eslint-disable @typescript-eslint/typedef */
+  const open = require('open');
+  const through = require('through2');
+  /* eslint-enable @typescript-eslint/typedef */
```
Comment (iclanton) [inline core-build/gulp-core-build-serve/src/utilities/GulpOpen.ts]:
```suggestion
  const open: typeof import('open') = require('open');
  const through: typeof import('through2') = require('through2');
```

File: core-build/gulp-core-build-serve/src/utilities/GulpOpen.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  opts = opts || {};
```
Comment (iclanton) [inline core-build/gulp-core-build-serve/src/utilities/GulpOpen.ts]: `opts` isn't optional.

File: core-build/gulp-core-build-serve/src/utilities/GulpOpen.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  /* eslint-disable-next-line @typescript-eslint/typedef */
+  return through.obj(function (file, enc, cb) {
+    let uri: string | undefined = opts.uri;
+
+    if (file.path && !uri) {
+      uri = file.path;
+    }
```
Comment (iclanton) [inline core-build/gulp-core-build-serve/src/utilities/GulpOpen.ts]:
```suggestion
    const uri: string | undefined = opts.uri || file.path;
```

**Review by iclanton — APPROVED** [review]

File: core-build/gulp-core-build-serve/src/utilities/GulpOpen.ts (hunk 1 above)
Comment (AJIXuMuK) [reply core-build/gulp-core-build-serve/src/utilities/GulpOpen.ts:4]: added

File: core-build/gulp-core-build-serve/src/utilities/GulpOpen.ts (hunk 2 above)
Comment (AJIXuMuK) [reply core-build/gulp-core-build-serve/src/utilities/GulpOpen.ts]: Adding types lead to build issues in `.d.ts` files of referenced modules.
Will leave as is.

File: core-build/gulp-core-build-serve/src/utilities/GulpOpen.ts (hunk 3 above)
Comment (AJIXuMuK) [reply core-build/gulp-core-build-serve/src/utilities/GulpOpen.ts]: addressed

File: core-build/gulp-core-build-serve/src/utilities/GulpOpen.ts (hunk 4 above)
Comment (AJIXuMuK) [reply core-build/gulp-core-build-serve/src/utilities/GulpOpen.ts]: addressed

**Review by iclanton — APPROVED** [review]

**Pull request merged**